legacy/core/src/main/java/com/fsck/k9/controller/MessagingController.java

```diff
@@ -2605,32 +2605,44 @@ public void checkAuthenticationProblem(LegacyAccountDto account) {
             handleAuthenticationFailure(account, true);
             return;
         } else {
-            clearAuthenticationErrorNotification(account, true);
+            clearAuthenticationErrorNotification(account, true, true);
         }
 
         // checking outgoing server configuration
         if (isAuthenticationProblem(account, false)) {
             handleAuthenticationFailure(account, false);
         } else {
-            clearAuthenticationErrorNotification(account, false);
+            clearAuthenticationErrorNotification(account, false, true);
         }
     }
 
     private boolean isAuthenticationProblem(LegacyAccountDto account, boolean incoming) {
-        ServerSettings serverSettings = incoming ?
-                account.getIncomingServerSettings() : account.getOutgoingServerSettings();
+        final ServerSettings serverSettings = getServerSettings(account, incoming);
 
         return serverSettings.isMissingCredentials() ||
                 serverSettings.authenticationType == AuthType.XOAUTH2 && account.getOAuthState() == null;
     }
 
-    private void clearAuthenticationErrorNotification(LegacyAccountDto account, boolean incoming) {
+    private ServerSettings getServerSettings(LegacyAccountDto account, boolean incoming) {
+        return incoming ? account.getIncomingServerSettings() : account.getOutgoingServerSettings();
+    }
+
+    private void clearAuthenticationErrorNotification(
+        LegacyAccountDto account, boolean incoming, boolean clearOnlyForOAuthAccounts) {
```

Comment:
This could be folded to the next line:


```suggestion
        LegacyAccountDto account, boolean incoming, boolean clearOnlyForOAuthAccounts
    ) {
```